internal/store/postgres/user_repository.go

```diff
@@ -277,14 +278,44 @@ func (r UserRepository) List(ctx context.Context, flt user.Filter) ([]user.User,
 
 	offset := (flt.Page - 1) * flt.Limit
 
-	query, params, err := dialect.From(TABLE_USERS).LeftOuterJoin(
-		goqu.T(TABLE_METADATA),
-		goqu.On(goqu.Ex{"users.id": goqu.I("metadata.user_id")})).Select("users.id", "name", "email", "key", "value", "users.created_at", "users.updated_at").Where(goqu.Or(
-		goqu.C("name").ILike(fmt.Sprintf("%%%s%%", flt.Keyword)),
-		goqu.C("email").ILike(fmt.Sprintf("%%%s%%", flt.Keyword)),
-	)).Limit(uint(flt.Limit)).Offset(uint(offset)).ToSQL()
-	if err != nil {
-		return []user.User{}, fmt.Errorf("%w: %s", queryErr, err)
+	var query string
+	var params []interface{}
+	var err error
+	if flt.Keyword != "" {
+		subQuery := dialect.From(TABLE_USERS).Where(goqu.Or(
+			goqu.C("name").ILike(fmt.Sprintf("%%%s%%", flt.Keyword)),
+			goqu.C("email").ILike(fmt.Sprintf("%%%s%%", flt.Keyword))),
+		)
+
+		query, params, err = dialect.From(subQuery.As(FILTERED_USERS)).LeftOuterJoin(
+			goqu.T(TABLE_METADATA),
+			goqu.On(goqu.Ex{fmt.Sprintf("%s.id", FILTERED_USERS): goqu.I(fmt.Sprintf("%s.user_id", TABLE_METADATA))})).Select(
+			fmt.Sprintf("%s.id", FILTERED_USERS),
+			fmt.Sprintf("%s.name", FILTERED_USERS),
+			fmt.Sprintf("%s.email", FILTERED_USERS),
+			fmt.Sprintf("%s.key", TABLE_METADATA),
+			fmt.Sprintf("%s.value", TABLE_METADATA),
+			fmt.Sprintf("%s.created_at", FILTERED_USERS),
+			fmt.Sprintf("%s.updated_at", FILTERED_USERS),
+		).Limit(uint(flt.Limit)).Offset(uint(offset)).ToSQL()
+		if err != nil {
+			return []user.User{}, fmt.Errorf("%w: %s", queryErr, err)
+		}
+	} else {
+		query, params, err = dialect.From(TABLE_USERS).LeftOuterJoin(
+			goqu.T(TABLE_METADATA),
+			goqu.On(goqu.Ex{fmt.Sprintf("%s.id", TABLE_USERS): goqu.I(fmt.Sprintf("%s.user_id", TABLE_METADATA))})).Select(
+			fmt.Sprintf("%s.id", TABLE_USERS),
+			fmt.Sprintf("%s.name", TABLE_USERS),
+			fmt.Sprintf("%s.email", TABLE_USERS),
+			fmt.Sprintf("%s.key", TABLE_METADATA),
+			fmt.Sprintf("%s.value", TABLE_METADATA),
+			fmt.Sprintf("%s.created_at", TABLE_USERS),
+			fmt.Sprintf("%s.updated_at", TABLE_USERS),
+		).Limit(uint(flt.Limit)).Offset(uint(offset)).ToSQL()
+		if err != nil {
+			return []user.User{}, fmt.Errorf("%w: %s", queryErr, err)
+		}
 	}
```

Comment:
Can we do this instead to shorten it?
```suggestion
  	var selectedUsers *goqu.SelectDataset

	if flt.Keyword != "" {
		selectedUsers = dialect.From(TABLE_USERS).Where(goqu.Or(
			goqu.C("name").ILike(fmt.Sprintf("%%%s%%", flt.Keyword)),
			goqu.C("email").ILike(fmt.Sprintf("%%%s%%", flt.Keyword))),
		).As(FILTERED_USERS)
	} else {
		selectedUsers = dialect.From(TABLE_USERS).As(FILTERED_USERS)
	}

	query, params, err = selectedUsers.LeftOuterJoin(
		goqu.T(TABLE_METADATA),
		goqu.On(goqu.Ex{fmt.Sprintf("%s.id", FILTERED_USERS): goqu.I(fmt.Sprintf("%s.user_id", TABLE_METADATA))})).Select(
		fmt.Sprintf("%s.id", FILTERED_USERS),
		fmt.Sprintf("%s.name", FILTERED_USERS),
		fmt.Sprintf("%s.email", FILTERED_USERS),
		fmt.Sprintf("%s.key", TABLE_METADATA),
		fmt.Sprintf("%s.value", TABLE_METADATA),
		fmt.Sprintf("%s.created_at", FILTERED_USERS),
		fmt.Sprintf("%s.updated_at", FILTERED_USERS),
	).Limit(uint(flt.Limit)).Offset(uint(offset)).ToSQL()
	if err != nil {
		return []user.User{}, fmt.Errorf("%w: %s", queryErr, err)
	}
```

Reply:
Yes we can